### DISCUSSION TITLE
timeouts on node command

### DISCUSSION THREAD

**Discussion by AdamBD** [starship/starship#5707] [Q&A] [answered]
Hi all,

I often (inconsistently) get this timeout when entering a directory with a package.json in it (I assume the presence of pacakge.json is what identifies this directory as a node one for the purposes of the prompt segment. 

Any idea what causes this?

[WARN] - (starship::utils): Executing command "node" timed out.
[WARN] - (starship::utils): You can set command_timeout in your config to a higher value to allow longer-running commands to keep executing.

**Comment by davidkna** [comment, marked as answer]
Starship runs `node --version` to determine the node versions for the `nodejs` module. This command can be slow to complete.

You can work around this by increasing the external command timeout, removing `$version` from the `nodejs` module format, or by disabling the module altogether.

**Reply by AdamBD** [reply]
@davidkna 

1. how would you change the external command timeout? You mean via the node cli configuration itself?
2. how would you remove the $version from the module format I dont see that as an option for the node module in the documentation: https://starship.rs/config/#node-js

**Reply by davidkna** [reply]
1. It's the top-level starship config option mentioned in the warning.
2. The format string can be set with the `format` option.
